### PR TITLE
updated README2.MD

### PR DESCRIPTION
**Pull request by heyzernut**

### CODE REVIEW

File: assets/js/script.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  $startBtn = $('.start')
+  $startBtn.on('click', () => {
+    document.getElementsByClassName('startpop')[0].pause()
+    stopKeyDown = true;
+    $('.start_pop').hide()
+    playBgSound ()
+    speed = 20
+  })
+
+	$container.on('mousedown', function () {
```
Comment (primaulia) [inline assets/js/script.js:69]: indentation issues here

File: assets/js/script.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      cPolePos = polePos
+    }
+    // move the poles
+    pole.css('right', cPolePos + speed)
+  }
+
+  function birdFlap () {
+    if (gameState === 1 || gameState === 2) {
+      $bird.css('transform', 'rotate(-20deg)')
+      $bird.stop().animate({
```
Comment (primaulia) [inline assets/js/script.js:142]: Why do you use animate method instead of doing this with [`css animation`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation) property instead?

File: assets/js/script.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    document.getElementsByClassName('hit')[0].play()
+  }
+  // audio
+
+  spop()
+
+  // start instruction popup
+  $startBtn = $('.start')
+  $startBtn.on('click', () => {
+    document.getElementsByClassName('startpop')[0].pause()
```
Comment (primaulia) [inline assets/js/script.js:62]: Try not to combine native dom manipulation method (`i.e. document getElementsByClassName`) with jquery method. Stick to one so you don't confuse yourself and anyone viewing your code

File: assets/js/script.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    document.getElementsByClassName('play_bgsound')[0].volume = 0.8
+    document.getElementsByClassName('play_bgsound')[0].play()
+  }
+
+  function hitOver() {
+    document.getElementsByClassName('hit')[0].play()
+  }
+  // audio
+
+  spop()
```
Comment (primaulia) [inline assets/js/script.js:57]: why this method is not called upon start of the game?

File: assets/js/script.js [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+  var birdPosInterval = setInterval(function () {
+    collisionDetection()
+  }, 50)
+
+  // collision session
+
+  // collision with pole
+  function collisionDetection () {
+    if ($bird.position().left < poleTop.position().left + poleTop.width() && $bird.position().left + $bird.width() > poleTop.position().left && $bird.position().top < poleTop.position().top + poleTop.height() && $bird.height() + $bird.position().top > poleTop.position().top) {
```
Comment (primaulia) [inline assets/js/script.js:98]: hmm can you please explain the different between these two collision detection?

File: assets/js/script.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+  function gravity () {
+    $bird.stop().animate({
+      bottom: '0'
+    })
+    $bird.css('transform', 'rotate(40deg)')
+  }
+
+  function gameEnd () {
+    gameState = 0
```
Comment (primaulia) [inline assets/js/script.js:165]: It seems that at the start of the code, you mentioned that your game depends on a certain game state. But I don't see any checking on the `gameState` at all.

**Comment by primaulia** [comment]
Project Workflow: 3 / 5
Technical Requirements: 3 / 5
Creativity: 3 / 5
Code Quality: 3 / 5
Problem Solving: 3 / 5
Problem Solving: 3 / 5

## Glow
- Nice design and overall feel
- Clone flappy bird nicely too

## Grow
- Lacking of modification in terms of game logic, definitely there's something else that you can think of 
- Code flow is messy, It's quite hard to follow your function flow
- Code repetition is apparent, but saw some refactoring
- Variable is quite clear, though it gets blurry at some point
- Focus less on design and focus more on rectifying your logic and make it your own

## Things to look out for
- Timeline; I still see updates after code freeze
- Priority in terms of developing your project
- Refactoring codes
- Combining native and jquery dom manipulation

File: assets/js/script.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+  var scoreAdd = false
+  var stopKeyDown = false;
+
+  function start () {
+    birdFlap()
+  }
+
+  // audio
+  function spop() {
+    document.getElementsByClassName('startpop')[0].play()
```
Comment (AlexMin314) [inline assets/js/script.js]: If you try $('.startpop').play(), it won't work. Do you understand why that doesn't work and how to do that in jQuery?

File: assets/js/script.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+  }
+
+  function flapsound() {
+    // document.getElementsByClassName('birdFlapping')[0].currentTime = 1.3
+    document.getElementsByClassName('birdFlapping')[0].pause()
+    document.getElementsByClassName('birdFlapping')[0].play()
```
Comment (alexkimin) [inline assets/js/script.js:44]: You already stored some dom accessing values into variables. (e.g. line 13 ~ )
What will be the benefit of doing that, and what can you do on line 43, 43?

File: assets/js/script.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        if (gameState > 1) return
+        if (gameState === 2) gameState = 1
+      }
+    }
+  })
+
+
+  var polePosInterval = setInterval(poleMove, 50)
+
+  var birdPosInterval = setInterval(function () {
```
Comment (alexkimin) [inline assets/js/script.js:90]: What will happen if I do like below? and why?

`var birdPosInterval = setInterval(collisionDetection, 50)`

File: assets/js/script.js [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+  }, 50)
+
+  // collision session
+
+  // collision with pole
+  function collisionDetection () {
+    if ($bird.position().left < poleTop.position().left + poleTop.width() && $bird.position().left + $bird.width() > poleTop.position().left && $bird.position().top < poleTop.position().top + poleTop.height() && $bird.height() + $bird.position().top > poleTop.position().top) {
+      gameEnd()
+    }
+    if ($bird.position().left < poleBottom.position().left + poleBottom.width() && $bird.position().left + $bird.width() > poleBottom.position().left && $bird.position().top < poleBottom.position().top + poleBottom.height() && $bird.height() + $bird.position().top > poleBottom.position().top) {
```
Comment (alexkimin) [inline assets/js/script.js:101]: Is there any way to reduce the number of calling that `.position()` method?

File: assets/js/script.js [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+  }
+
+  // audio
+  function spop() {
+    document.getElementsByClassName('startpop')[0].play()
+  }
+
+  function flapsound() {
+    // document.getElementsByClassName('birdFlapping')[0].currentTime = 1.3
+    document.getElementsByClassName('birdFlapping')[0].pause()
```
Comment (AlexMin314) [inline assets/js/script.js]: If you try `$('.startpop').play()`, it won't work. Do you understand why that doesn't work and how to make that works in jQuery?

Comment (alexkimin) [inline assets/js/script.js:43]: If you try `$('.startpop').play()`, it won't work. Do you understand why that doesn't work and how to make that works in jQuery?

**Comment by alexkimin** [comment]
Project Workflow: 4 / 5
Technical Requirements: 3 / 5
Creativity: 3 / 5
Code Quality: 3 / 5
Problem Solving: 3 / 5


## Glow

Nice design and fun game.

## Grow

Maybe you could do more features and logics before juicing up design.
If your pole generation were done with `Class`(constructor function), it would be better.

Check this. 
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes](url)

## Things to look out for

Refactoring codes
Combining native and jquery dom manipulation
